tests/gdmem/malloc.c

```diff
@@ -0,0 +1,77 @@
+#include "gd.h"
+#include "gdtest.h"
+
+void *testcalloc(size_t nmemb, size_t size)
+{
+	char *ptr;
+
+	ptr = calloc(nmemb, size);
+
+	for (size_t i = 0; i < size; i++)
+		*(ptr + i) = 0;
+
+	return (void *)ptr;
+}
+
+void *testmalloc(size_t size)
+{
+	char *ptr;
+
+	ptr = malloc(size);
+
+	for (size_t i = 0; i < size; i++)
+		*(ptr + i) = 0;
```

Comment:
`malloc()` returns uninitialized memory, not zero-ed memory.  if you're trying to poison things, maybe use 0xaa or something ?